var/spack/repos/builtin/packages/openssh/package.py

```diff
@@ -54,6 +55,8 @@ class Openssh(AutotoolsPackage):
 
     depends_on("krb5+shared", when="+gssapi")
     depends_on("openssl@:1.0", when="@:7.7p1")
+    depends_on("openssl@:1.1", when="@:7.9p1")
+    depends_on("openssl@:3.0", when="@:8.7p1")
```

Comment:
Probably you mean `@:3` here? OpenSSL 3 follows semver.

Reply:
**Nope, semver has nothing to with that:**

OpenSSH-8.7 works with OpenSSL 3.0 but don't support 3.1.

Reply:
Then I guess it's a faulty configure script, like with zlib, but let's leave it :)